contracts/bonding/libraries/EarningsPoolV2.sol

```diff
@@ -0,0 +1,306 @@
+        uint256 cumulativeRewardFactor; // cumulative reward factor since using LIP36
+        uint256 cumulativeFeeFactor; // cumulative fee factor since using LIP36
+    }
+
+    /**
+     * @notice Sets transcoderRewardCut and transcoderFeeshare for an EarningsPool
+     * @param earningsPool Storage pointer to EarningsPool struct
+     * @param _rewardCut Reward cut of transcoder during the earnings pool's round
+     * @param _feeShare Fee share of transcoder during the earnings pool's round
+     */
+    function setCommission(EarningsPool.Data storage earningsPool, uint256 _rewardCut, uint256 _feeShare) internal {
+        earningsPool.transcoderRewardCut = _rewardCut;
+        earningsPool.transcoderFeeShare = _feeShare;
+    }
+
+    /**
+     * @notice Sets totalStake for an EarningsPool
+     * @param earningsPool Storage pointer to EarningsPool struct
+     * @param _stake Total stake of the transcoder during the earnings pool's round
+     */
+    function setStake(EarningsPool.Data storage earningsPool, uint256 _stake) internal {
+        earningsPool.totalStake = _stake;
+    }
+
+    /**
+     * @notice Return whether this earnings pool has claimable shares i.e. is there unclaimed stake
+     * @dev DEPRECATED - for backwards compatibility with v1 EarningsPools
+     * @param earningsPool Storage pointer to EarningsPool struct
+     */
+    function hasClaimableShares(EarningsPool.Data storage earningsPool) internal view returns (bool) {
+        return earningsPool.claimableStake > 0;
+    }
+
+    /**
+     * @notice Add fees to the earnings pool
+     * @param earningsPool Storage pointer to EarningsPools struct
+     * @param _fees Amount of fees to add
+     */
+    function addToFeePool(EarningsPool.Data storage earningsPool, EarningsPool.Data storage _prevEarningsPool, uint256 _fees) internal {
+        uint256 prevCumulativeFeeFactor = _prevEarningsPool.cumulativeFeeFactor;
+        uint256 prevCumulativeRewardFactor = _prevEarningsPool.cumulativeRewardFactor != 0 ? _prevEarningsPool.cumulativeRewardFactor : MathUtils.percPoints(1,1);
+
+        if (earningsPool.cumulativeFeeFactor == 0) {
+            earningsPool.cumulativeFeeFactor = prevCumulativeFeeFactor.add(
+                MathUtils.percOf(prevCumulativeRewardFactor, _fees, earningsPool.totalStake)
+            );
+            return;
+        }
+
+        earningsPool.cumulativeFeeFactor = earningsPool.cumulativeFeeFactor.add(
+                MathUtils.percOf(prevCumulativeRewardFactor, _fees, earningsPool.totalStake)
+            );
+    }
+
+    /**
+     * @dev Add rewards to the earnings pool
+     * @param earningsPool Storage pointer to EarningsPool struct
+     * @param _rewards Amount of rewards to add
+     */
+    function addToRewardPool(EarningsPool.Data storage earningsPool, EarningsPool.Data storage _prevEarningsPool, uint256 _rewards) internal {
```

Comment:
We may be able to reduce the diff of this PR and more cleanly separate the LIP-36 code by doing something like the following:

- Keep the existing `EarningsPool.sol` file
    - Add the new fields to the `EarningsPool` struct
    - Remove the `addToRewardPool()` and `addToFeePool()` functions from the file
- Create a new file `EarningsPoolLIP36.sol` which would look like:

```solidity
import "./EarningsPool.sol";

library EarningsPoolLIP36 {
    function addToRewardPool(EarningsPool.Data storage earningsPool, ...) { ... }
    function addToFeePool(EarningsPool.Data storage earningsPool, ...) { ... }
}
```

Then, using the new library in the BondingManager would look something like:

```solidity
import "./libraries/EarningsPool.sol";
import "./libraries/EarningsPoolLIP36.sol";

contract BondingManager {
    ...
    using EarningsPool for EarningsPool.Data;
    using EarningsPoolLIP36 for EarningsPool.Data;
}
```

WDYT?

Reply:
Addressed in dff6dd2